hail/src/main/scala/is/hail/expr/types/physical/PSubsetStruct.scala

```diff
@@ -0,0 +1,225 @@
+package is.hail.expr.types.physical
+
+import is.hail.annotations.{Region, UnsafeUtils}
+import is.hail.asm4s.{Code, Settable, SettableBuilder, Value, const}
+import is.hail.expr.ir.{EmitCodeBuilder, EmitMethodBuilder, IEmitCode}
+import is.hail.expr.types.BaseStruct
+import is.hail.expr.types.virtual.TStruct
+import is.hail.utils._
+
+object PSubsetStruct {
+  def apply(ps: PStruct, fieldNames: String*): PSubsetStruct = {
+    val f = fieldNames.map(f => ps.field(f)).toFastIndexedSeq
+    PSubsetStruct(ps, f)
+  }
+}
+
+// PSubsetStruct is a view of some other PStruct
+// Operations it can take:
+// 1) Modify the view by changing which fields from the backing struct it has access to (add/delete, but don't modify backing PStruct)
+// 2) Modify the backing values (setting requiredeness)
+// 3) Modify properties of the fields in the view (rename), which requires the generation of a new backing PStruct
+// Creating entirely new fields seems out of scope
+final case class PSubsetStruct(ps: PStruct, fields: IndexedSeq[PField]) extends PStruct {
```

Comment:
Yeah, I hadn't because requires an override on fieldNames (and still need to construct fields), and fieldNames would need to be taken as an Array, which is a bit unusual (we typically pass around IndexedSeq), so a bit of a messier implementation. In any case, I've done as you asked, and we can always play around with the interface later.